lib/stdlib/src/binary.erl

```diff
@@ -70,16 +85,26 @@ bin_to_list(Subject, Pos, Len) when not is_binary(Subject);
     %% binary_to_list/3 allows bitstrings as long as the slice fits, and we
     %% want to badarg when Pos/Len aren't integers instead of raising badarith
     %% when adjusting args for binary_to_list/3.
-    erlang:error(badarg);
+    badarg_with_info([Subject, Pos, Len]);
 bin_to_list(Subject, Pos, 0) when Pos >= 0, Pos =< byte_size(Subject) ->
     %% binary_to_list/3 doesn't handle this case.
     [];
-bin_to_list(_Subject, _Pos, 0) ->
-    erlang:error(badarg);
 bin_to_list(Subject, Pos, Len) when Len < 0 ->
-    bin_to_list(Subject, Pos + Len, -Len);
+    try
+        bin_to_list(Subject, Pos + Len, -Len)
+    catch
+        error:Reason ->
+            error_with_info(Reason, [Subject, Pos, Len])
```

Comment:
I wonder if this will have correct stacktrace. Wouldn't this report the raising function as the `error_with_info`, rather than `bin_to_list` since it's a tail call

Reply:
The call will be inlined. See line 40.

Reply:
Ah, that makes sense. Sorry for the noise